lib/OpenQA/WebAPI/Controller/Admin/Needle.pm

```diff
@@ -17,33 +17,22 @@ sub index {
     $self->render('admin/needle/index');
 }
 
-sub _translate_days($) {
-    my ($days) = @_;
-    return time2str('%Y-%m-%d %H:%M:%S', time - $days * ONE_DAY, 'UTC');
-}
+sub _translate_days ($days) { time2str('%Y-%m-%d %H:%M:%S', time - $days * ONE_DAY, 'UTC') }
 
-sub _translate_date_format($) {
-    my ($datetime) = @_;
-    my $datetime_obj = DateTime::Format::Pg->parse_datetime($datetime);
-    return DateTime::Format::Pg->format_datetime($datetime_obj);
+sub _translate_date_format ($datetime) {
+    DateTime::Format::Pg->format_datetime(DateTime::Format::Pg->parse_datetime($datetime));
 }
 
-sub _translate_cond($) {
-    my ($cond) = @_;
-
-    if ($cond =~ m/^min(\d+)$/) {
-        return {'>=' => _translate_days($1)};
-    }
-    elsif ($cond =~ m/^max(\d+)$/) {
-        return [{'<' => _translate_days($1)}, {'=' => undef}];
+sub _translate_cond ($cond) {
+    my ($operator, @additional_conds) = rindex($cond, 'min', 0) == 0 ? ('>=') : ('<', {'=' => undef});
```

Comment:
Why are you using `rindex` here instead of `index`?

Reply:
I actually just want a `starts_with`. I suppose neither `index` nor `rindex` are providing that directly but `rindex` would maybe waste less time going though the whole string if `min` in not in the string.

Reply:
Anchored perl regexes are really fast, plus this code isn't really performance critical.
So I would suggest `$cond =~ m/^min/`

Reply:
I have changed it to use a regex. Maybe that is better readable for more people then. (However, I don't think it is generally more readable. It really depends on whether one knows regex syntax or the rindex function better. And of course what we'd really wanted would be an intuitively named function like `starts_with`.)